tests/h/cli/commands/user_test.py

```diff
@@ -62,266 +58,190 @@ def test_it_adds_user_with_specific_authority(self, cli, cliconfig, signup_servi
+        self, invoke_cli, user, db_session, user_password_service
     ):
-        result = cli.invoke(
-            user_cli.password, [user.username, "--password", "newpass"], obj=cliconfig
-        )
+        result = invoke_cli(user_cli.password, [user.username, "--password", "newpass"])
 
         assert not result.exit_code
 
-        user = db_session.query(models.User).get(user.id)
-        assert password_service.check_password(user, "newpass")
+        user = db_session.query(User).get(user.id)
+        user_password_service.update_password.assert_called_once_with(user, "newpass")
 
     def test_it_changes_password_with_specific_authority(
-        self, cli, cliconfig, user, db_session, password_service
+        self, invoke_cli, user, db_session, user_password_service
     ):
         user.authority = "partner.org"
         db_session.flush()
 
-        result = cli.invoke(
+        result = invoke_cli(
             user_cli.password,
             ["--authority", "partner.org", user.username, "--password", "newpass"],
-            obj=cliconfig,
         )
 
         assert not result.exit_code
 
-        user = db_session.query(models.User).get(user.id)
-        assert password_service.check_password(user, "newpass")
+        user = db_session.query(User).get(user.id)
+        user_password_service.update_password.assert_called_once_with(user, "newpass")
 
     def test_it_errors_when_user_could_not_be_found(
-        self, cli, cliconfig, user, db_session, password_service
+        self, invoke_cli, user_password_service
     ):
-        result = cli.invoke(
-            user_cli.password,
-            [f"bogus_{user.username}", "--password", "newpass"],
-            obj=cliconfig,
+        result = invoke_cli(
+            user_cli.password, ["bogus_username", "--password", "newpass"]
         )
 
         assert result.exit_code == 1
 
-        user = db_session.query(models.User).get(user.id)
-        assert not password_service.check_password(user, "newpass")
+        user_password_service.update_password.assert_not_called()
 
     def test_it_errors_when_user_with_specific_authority_could_not_be_found(
-        self, cli, cliconfig, user, db_session, password_service
+        self, invoke_cli, user, user_password_service
     ):
-        result = cli.invoke(
+        result = invoke_cli(
             user_cli.password,
             ["--authority", "foo.com", user.username, "--password", "newpass"],
-            obj=cliconfig,
         )
 
         assert result.exit_code == 1
 
-        user = db_session.query(models.User).get(user.id)
-        assert not password_service.check_password(user, "newpass")
+        user_password_service.update_password.assert_not_called()
 
     @pytest.fixture
-    def user(self, db_session, factories):
-        user = factories.User()
-        db_session.flush()
-        return user
+    def user(self, factories):
+        return factories.User()
```

Comment:
We could probably pull this out as a shared fixture, but it's so small I don't really mind the duplication here.